src/Microsoft.Identity.Client/UI/IWebUI.cs

```diff
@@ -19,8 +19,9 @@ internal interface IWebUI
         /// <summary>
         /// Extra validations on the redirect uri, for example system web views cannot work with the urn:oob... uri because
         /// there is no way of knowing which app to get back to.
+        /// WebUIs can update the uri, for example use http://localhost:1234 instead of http://localhost
         /// Throws if uri is invalid
         /// </summary>
-        void ValidateRedirectUri(Uri redirectUri);
+        Uri ValidateRedirectUri(Uri redirectUri);
```

Comment:
I am a bit unsure about how clean this implementation is: 

- the name of this method is probably not descriptive enough. Maybe `UpdateOrValidate` or smth? Most webUIs do only validation;

Reply:
Also not sure about returning the same as the input... Validate indicates more like returning a bool, if it updates the returnurl.... method should reflect that

---
In reply to: [284316881](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1150#discussion_r284316881) [](ancestors = 284316881)

Reply:
Renamed to UpdateRedirectUri